### DISCUSSION TITLE
How to specify ignoreRecord in unique rule in Import?

### DISCUSSION THREAD

**Discussion by skills-up** [filamentphp/filament#11755] [Help] [answered]
### Package

Actions

### Package Version

v3.2.45

### How can we help you?

I've added an import action to a model with option to update existing record via `firstOrNew` method. Now, I wish to have the unique rule applied on another column, but only when the record is not being updated with the same value.

Typical laravel way for this is to ignore record with matching id. However, how do I specify this in the `rules()` parameter for a column in `getColumns` method of an Importer?

**Comment by naseef** [comment, marked as answer]
`->rules(fn ($record) => [Rule::unique('users')->ignore($record->id)])`

**Reply by ariesdg** [reply]
Hi, did you manage resolve this?

**Reply by skills-up** [reply]
Yes, the above answer by @naseef worked.
